python/cudf/cudf/core/column/column.py

```diff
@@ -1271,11 +1271,13 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             )
 
         elif isinstance(arbitrary, pa.NullArray):
-            new_dtype = pd.api.types.pandas_dtype(dtype)
             if (type(dtype) == str and dtype == "empty") or dtype is None:
                 new_dtype = pd.api.types.pandas_dtype(
                     arbitrary.type.to_pandas_dtype()
                 )
+            else:
+                new_dtype = pd.api.types.pandas_dtype(dtype)
```

Comment:
This fixes a previously undetected issue reading en empty index from a parquet file. In pandas 0.25.3, inside our fixture that creates the test data, the index was being inadvertently dropped (converted to a RangeIndex) when what we really wanted was an empty index named `test_index` with an `empty` datatype. This was fixed apparently in pandas 1.0 meaning the parquet file actually contains the desired empty index, and this revealed an error in our datatype handling within as_column - basically `pd.api.types.pandas_dtype` can't handle "empty".